Code/autopkglib/FileFinder.py

```diff
@@ -96,6 +97,15 @@ def main(self):
                 self.output(
                     f"DMG-relative file match: '{self.env['dmg_found_filename']}'"
                 )
+
+            if match.endswith('/'):
+                self.env["found_basename"] = os.path.basename(match.lstrip("/"))
```

Comment:
I'm not really sure I understand the purpose of this check. When would the match end with /?

Reply:
@nmcspadden this was just in case someone passed a folder path to FileFinder, as per the attached.

The below is an example test:

```
#!/usr/local/autopkg/python

import os.path
from glob import glob

pattern = '/Users/ben/Library/AutoPkg/Cache/ch.unibas.its.git.mcs.munki.tableau-prep/payload/Tableau*.app'
glob_matches = glob(pattern, recursive=True)
glob_matches.sort()

match = glob_matches[-1]

if match.endswith('/'):
	found_basename = os.path.basename(match.lstrip("/"))
else:
	found_basename = os.path.basename(match)
print(
	f"Basename match .app: '{found_basename}'"
)

print ()

pattern = '/Users/ben/Library/AutoPkg/Cache/ch.unibas.its.git.mcs.munki.tableau-prep/payload/'
glob_matches = glob(pattern, recursive=True)
glob_matches.sort()

match = glob_matches[-1]

if match.endswith('/'):
	found_basename = os.path.basename(match.rstrip("/"))
else:
	found_basename = os.path.basename(match)
print(
	f"Basename match dir: '{found_basename}'"
)
```

Output:

So, pattern input '/Users/ben/Library/AutoPkg/Cache/ch.unibas.its.git.mcs.munki.tableau-prep/payload/Tableau*.app' will output:

```Basename match .app: 'Tableau Prep Builder 2021.1.app'```

And pattern input '/Users/ben/Library/AutoPkg/Cache/ch.unibas.its.git.mcs.munki.tableau-prep/payload/' will output:

```Basename match dir: 'payload'```

From the above.. this us more of a convenience for folks tbh..

Oh and had to change the lstrip to rstrip.. which was the commit done just now